### PR TITLE
gtests: adds algebraic tests

### PR DESCRIPTION
**Pull request by PetholzA**
also changes a tiny copy&paste error in DenseMatrix.cpp

### CODE REVIEW

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 1]
```diff
@@ -508,6 +508,66 @@ TYPED_TEST(MatricesGTest, testMatrixAddition) {
...
 }
 
+TYPED_TEST(MatricesGTest, testMatrixAdditionEqual) {
+    const int num_triplets = 100;
+    std::vector<Triplet> triplets1(num_triplets);
+    std::vector<Triplet> triplets2(num_triplets);
+
+    for (index i = 0; i < num_triplets; ++i) {
+        triplets1[i] = {i, i, 1};
+        triplets2[i] = {i, i, (double)i};
```
Comment (fabratu) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:518]: Use `static_cast` for changing literal types.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 2]
```diff
@@ -828,6 +931,65 @@ TYPED_TEST(MatricesGTest, testMatrixMultiplication) {
     EXPECT_EQ(result(2, 1), 4);
 }
 
+TYPED_TEST(MatricesGTest, testMatricesTransposedMultiplication) {
+    using Matrix = typename TestFixture::Matrix;
+    if (std::is_same<Matrix, DenseMatrix>::value) {
+        GTEST_SKIP() << "Skipping mmT/mTm test for DenseMatrix";
+    }
+    std::vector<Triplet> triplets1 = {{0, 0, 1}, {0, 1, 2}, {0, 2, 3}, {1, 0, 2}, {1, 1, 2}};
+    //		     1  2  3
```
Comment (fabratu) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:940]: Indentation looks broken.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 3]
```diff
@@ -828,6 +931,65 @@ TYPED_TEST(MatricesGTest, testMatrixMultiplication) {
...
+    // mat =	 2  2  0
+    //
+    DynamicMatrix mat(2, 3, triplets1);
+    ASSERT_EQ(mat.numberOfRows(), 2);
+    ASSERT_EQ(mat.numberOfColumns(), 3);
+    //		     1  2
```
Comment (fabratu) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:946]: As above.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 4]
```diff
@@ -828,6 +931,65 @@ TYPED_TEST(MatricesGTest, testMatrixMultiplication) {
...
+    // mat^T =	 2  2
+    //           3  0
+    DynamicMatrix result = DynamicMatrix::mmTMultiply(mat, mat);
+    // (mat*mat^T)
+    //			14  6
```
Comment (fabratu) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:951]: As above.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp [hunk 5]
```diff
@@ -828,6 +931,65 @@ TYPED_TEST(MatricesGTest, testMatrixMultiplication) {
...
+    // result = 6   8
+    EXPECT_EQ(result(0, 0), 14);
+    EXPECT_EQ(result(0, 1), 6);
+    EXPECT_EQ(result(1, 0), 6);
+    EXPECT_EQ(result(1, 1), 8);
+
+    result = DynamicMatrix::mTmMultiply(mat, mat);
+    // (mat^T*mat)
+    //			5  6  3
```
Comment (fabratu) [inline networkit/cpp/algebraic/test/MatricesGTest.cpp:960]: As above.

**Review by fabratu — CHANGES_REQUESTED** [review]
Good catch with the error in subtraction.

In addition, check whether the `ToDo`s near the start of `MatricesGTest.cpp` can be deleted.

File: networkit/cpp/algebraic/test/MatricesGTest.cpp (hunk 2 above)
Comment (PetholzA) [reply networkit/cpp/algebraic/test/MatricesGTest.cpp:936]: indentation looks as intended in VScode (displaying the matrices/vectors visually)

**Comment by fabratu** [comment]
The problem concerning indentation is introduced by mixing `tabs` and `spaces`. While VSCode does try to resolve this, other editors often do not (and also VSCode sometimes fails). Please resolve this by only using spaces.

**Comment by PetholzA** [comment]
> The problem concerning indentation is introduced by mixing `tabs` and `spaces`. While VSCode does try to resolve this, other editors often do not (and also VSCode sometimes fails). Please change the indentation to only use `spaces`.

should be addressed

**Review by fabratu — APPROVED** [review]
LGTM now.

**Pull request merged**